src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,290 @@
+        String taskDescription = String.valueOf(tasks.get(deleteIndex));
+        Task toDelete = tasks.get(deleteIndex);
+        int taskLeft = tasks.get(0).getNumberOfTasks() - 1;
+        toDelete.remove();
+        tasks.remove(deleteIndex);
+        System.out.println(DELETED_THIS_TASK + "\n" + taskDescription + "\nNow you have " + taskLeft + " tasks in the list.");
+    }
+
+    private static void checkIfCommandEmpty(String[] words) throws EmptyCommandException {
+        if (words[0].equals(LIST_COMMAND)) {
+            return;
+        }
+        if (words.length < 2 || words[1].equals("")) {
+            throw new EmptyCommandException();
+        }
+    }
+
+    private static Todo getTodo(String[] words) {
+        Todo td = new Todo(words[1]);
+        return td;
+    }
+
+    private static void printList() {
+        int listIndex = 0;
+        for (Task t : tasks) {
+            if (t != null) {
+                System.out.println(++listIndex + ". " + t);
+            }
+        }
+    }
+
+
+    private static void writeTasksToFile(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task t : tasks) {
+            if (t != null) {
+                fw.write(t.saveFormat() + System.lineSeparator());
+            }
+        }
+
+        fw.close();
+    }
+
+    private static void save(String filename) throws IOException {
+        try {
+            writeTasksToFile(filename);
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void load(String filename) {
+        File f = new File(filename);
+
+        if (f.exists()) {
+            try {
+                loadFileContents(f);
+            } catch (FileNotFoundException e) {
+                System.out.println(FILE_NOT_FOUND_MESSAGE);
+            }
+        }
+    }
+
+    private static void loadFileContents(File f) throws FileNotFoundException {
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            processFileContents(s.nextLine());
+        }
+    }
+
+    private static void processFileContents(String line){
+        String[] words = line.split("\\|");
+        String type = words[0];
+        boolean isDone = false;
+        if (Integer.valueOf(words[1]) == 1) {
+            isDone = true;
+        }
+        switch (type) {
+        case "T":
+            loadTodo(words, isDone);
+            break;
+        case "E":
+            loadEvent(words, isDone);
+            break;
+        case "D":
+            loadDeadline(words, isDone);
+            break;
+        default:
+            // unknown char error
+        }
+    }
+
+    private static void loadDeadline(String[] words, boolean isDone) {
+        Deadline deadline = new Deadline(words[2], words[3]);
+        if(isDone){
```

Comment:
```suggestion
        if (isDone) {
```
Similar for other occurrences.